File: Data Structures/Arrays/Cpp/Search_in_a_matrix.cpp
// Write a program to read an array and search for an number if it is present in the matrix or not.

#include<iostream>
using namespace std;

int main(){
    int row, col;
    cout << "Enter the number of rows for Matrix :" << " ";
    cin >> row;
    cout << "Enter the number of columns for Matrix :" << " ";
    cin >> col;
    int Matrix[row][col];
    cout << "Enter the elements of Matrix : " << endl;
    for(int i = 0; i < row; i++){
        for(int j = 0; j < col; j++){
            cin >> Matrix[i][j];
        }
    }
    int search;
    cout << "Enter the number you want to search in the Matrix:" << " ";
    cin >> search;
    bool flag = false;
    for(int i = 0; i < row; i++){
        for(int j = 0; j < col; j++){
            if (Matrix[i][j] == search){
                flag = true;
                break;
            }
        }
        if (flag == true){
            break;
        }
    }
    if (flag == true){
        cout << "Number found in the matrix." << endl;
    }
    else{
        cout << "Number not found in the matrix." << endl;
    }
    return 0;
}

File: Data Structures/Arrays/Java/Search_in_a_matrix.java
// Write a program to read an array and search for an number if it is present in the matrix or not.

import java.util.Scanner;

public class Search_in_a_matrix{
    public static void main(String[] args){
        Scanner read = new Scanner(System.in);
        int[][] Array = new int[3][3];
        for(int i = 0; i < 3; i++){
            for(int j = 0; j < 3; j++){
                Array[i][j] += read.nextInt();
            }
        }
        System.out.print("Enter the number you want to search in the matrix: ");
        int search = read.nextInt();
        boolean flag = false;
        for(int i = 0; i < 3; i++){
            for(int j = 0; j < 3; j++){
                if (Array[i][j] == search){
                    flag = true;
                    break;
                }
            }
            if (flag == true){
                break;
            }
        }
        if (flag == true){
            System.out.println("The number is present in the matrix.");
        }
        else{
            System.out.println("The number is not present in the matrix.");
        }
        read.close();
    }
}